File: lecture005/reverse.java
import java.util.Scanner;

public class reverse {
    public static Scanner scn = new Scanner(System.in);

    public static void main(String[] args) {

        // int n;
        // n=scn.nextInt();

        System.out.print("Enter si : ");
        int si = scn.nextInt();
        System.out.print("Enter ei : ");
        int ei = scn.nextInt();
        System.out.print("Enter no of elements in array : ");
        int n;
        n = scn.nextInt();
        // System.out.println();
        int[] arr = new int[n];
        // System.out.println();
        input(arr);
        // System.out.println();
        reverse_(arr, si, ei);
        // System.out.println();
        display(arr);
    }

    public static void display(int[] arr) {
        for (int i = 0; i < arr.length; i++)
            System.out.print(arr[i] + " ");
    }

    public static void swap(int[] arr, int y, int x) {
        int temp = arr[x];
        arr[x] = arr[y];
        arr[y] = temp;
    }

    public static void reverse_(int[] arr, int si, int ei) {
        while (si < ei) {
            swap(arr, si, ei);
            si++;
            ei--;
        }
    }

    public static void input(int[] arr) {
        for (int i = 0; i < arr.length; i++)
            arr[i] = scn.nextInt();
    }

}

File: lecture005/reverse.cpp
#include <iostream>
#include <vector>
using namespace std;

void input(vector<int> &arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

void swap(vector<int> &arr, int y, int x)
{
    int temp = arr[x];
    arr[x] = arr[y];
    arr[y] = temp;
}

void reverse_(vector<int> &arr, int si, int ei)
{
    while (si < ei)
    {
        swap(arr, si, ei);
        si++;
        ei--;
    }
}

void display(vector<int> &arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        cout << arr[i] << " ";
    }
}

int main(int args, char **argv)
{
    int a, n;
    cout << "Enter no of elements in the array : ";
    cin >> n;
    vector<int> arr(n, 0);
    cout << "Enter an array : ";
    input(arr);
    int si;
    cout << "Enter starting index : ";
    cin >> si;
    int ei;
    cout << "Enter end index : ";
    cin >> ei;
    reverse_(arr, si, ei);
    cout << "reversed array is : ";
    display(arr);
    return 0;
}
